cachi2/core/package_managers/pip.py

```diff
@@ -1482,12 +1488,15 @@ def _hash_verify(path: Path, checksum_info: Iterable[ChecksumInfo]) -> bool:
         if dpi.package_type == "sdist":
             _check_metadata_in_sdist(dpi.path)
 
+        download_info["package_type"] = dpi.package_type
+
     else:
         if require_hashes or req.kind == "url":
             hashes = req.hashes or [req.qualifiers.get("cachito_hash", "")]
             download_info["hash_verified"] = _hash_verify(
                 download_info["path"], list(map(_to_checksum_info, hashes))
             )
+            download_info["package_type"] = ""  # hack
```

Comment:
Not really happy with this, but given that DPIs only exist for pypi deps, it doesn't make sense (to me) to add "", or "none", or some other null to the `DistributionPackageInfo::package_type` Literal

Reply:
Shouldn't be this `download_info["package_type"] = "sdist"` ?

(on the same level as `if ...` )

Reply:
No, otherwise we hit the situation where VCS/URL reqs **don't** have the "package_type" property _at all_.

Reply:
But now that I'm looking at it again, I think that VCS reqs aren't covered...

Reply:
> Shouldn't this be ... on the same level as `if`

I shouldn't review these comments first thing in the morning :roll_eyes: - you're absolutely correct

Reply:
It shouldn't be "sdist" for VCS/URL reqs, because **they aren't sdists** - "sdist"/"wheel" only apply to downloads from PyPi.

The empty string is the "hack", but I don't know of a better workaround that wouldn't require a substantial refactor.

Reply:
ACK